packages/internal-test-helpers/lib/test-cases/abstract-application.js

```diff
@@ -59,8 +59,10 @@ export default class AbstractApplicationTestCase extends AbstractTestCase {
     if (applicationInstance) {
       return run(applicationInstance, 'visit', url, options);
     } else {
-      return run(this.application, 'visit', url, options).then(instance => {
-        this.applicationInstance = instance;
```

Comment:
b/c this `.then` is not inside he runloop, `this.applicationInstance` was being assigned *after* `this.visit('/foo')`. The more predictable behavior is to wrap the `then` in a runloop so we can presume the instance is set on the next line of code after `this.visit`